app/qml/components/KeyboardPage.qml

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2019 Lieven Hey <t.schilling@snafu.de>.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+
+Page {
+    id: page
+    allowedOrientations: Orientation.All
+    property var device
+    property var plugin: device.plugin("SailfishConnect::RemoteKeyboardPlugin")
+    property QtObject modifiers: QtObject {
+        property int shift: 0
+        property int ctrl: 0
+        property int alt: 0
+    }
+    property bool chars: false
+
+    function displayCorrectChar(data) {
+        if (page.chars && typeof data["symView"] !== "undefined") {
+            // dont show text on arrow keys
+            if (["up", "down", "left", "right"].indexOf(data["symView"]) != -1) return " ";
+            if (modifiers.shift && typeof data["symView2"] !== "undefined") {
+                return data["symView2"];
+            }
+            return data["symView"];
+        } else {
+            if (modifiers.shift && typeof data["captionShifted"] !== "undefined") {
+                return data["captionShifted"];
+            }
+            return data["caption"];
+        }
+    }
+
+    PageHeader {
```

Comment:
Please use as header of SilicaFlickable:
```
SilicaFlickable {
  header: PageHeader { ...
```

Reply:
Does not work: Error: Error while loading page: file:///usr/share/harbour-sailfishconnect/qml/components/KeyboardPage.qml:52 Cannot assign to non-existent property "header"